src/system/flow.rs

```diff
@@ -27,74 +27,110 @@ use async_std::{
     task,
 };
 use hashbrown::HashMap;
-use std::{borrow::Borrow, collections::HashSet};
+use std::collections::HashSet;
 use std::{sync::atomic::Ordering, time::Duration};
 use tremor_common::ids::{ConnectorIdGen, OperatorIdGen};
 use tremor_script::{
-    ast::{self, ConnectStmt, DeployEndpoint, DeployFlow, Helper},
+    ast::{self, ConnectStmt, DeployFlow, Helper},
     errors::not_defined_err,
 };
 
 /// unique identifier of a flow instance within a tremor instance
-#[derive(Debug, PartialEq, PartialOrd, Eq, Hash, Clone, Serialize)]
-pub(crate) struct Id(pub(crate) String);
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, PartialEq, PartialOrd, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct FlowAlias(String);
+
+impl FlowAlias {
+    /// construct a new flow if from some stringy thingy
+    pub fn new(alias: impl Into<String>) -> Self {
+        Self(alias.into())
+    }
 
-impl From<&DeployFlow<'_>> for Id {
+    /// reference this id as a stringy thing again
+    #[must_use]
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl From<&DeployFlow<'_>> for FlowAlias {
     fn from(f: &DeployFlow) -> Self {
-        Id(f.instance_alias.to_string())
+        FlowAlias(f.instance_alias.to_string())
     }
 }
 
-impl From<&str> for Id {
+impl From<&str> for FlowAlias {
     fn from(e: &str) -> Self {
-        Id(e.to_string())
+        FlowAlias(e.to_string())
+    }
+}
+
+impl std::fmt::Display for FlowAlias {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
     }
 }
 
 /// unique instance alias/id of a connector within a deployment
 #[derive(Debug, PartialEq, PartialOrd, Eq, Hash, Clone, Serialize, Deserialize)]
-pub struct ConnectorAlias(String);
+pub struct ConnectorAlias {
+    flow_alias: FlowAlias,
+    connector_alias: String,
+}
 
-impl From<&DeployEndpoint> for ConnectorAlias {
-    fn from(e: &DeployEndpoint) -> Self {
-        ConnectorAlias(e.alias().to_string())
+impl ConnectorAlias {
+    /// construct a new `ConnectorId` from the id of the containing flow and the connector instance id
+    pub fn new(flow_alias: FlowAlias, connector_alias: impl Into<String>) -> Self {
+        Self {
+            flow_alias,
+            connector_alias: connector_alias.into(),
+        }
     }
```

Comment:
Changing the new definition would make a lot of the `new` calls easier through the files as they could be passed something like `ConnectorAlias::new("flow", "connector")` in the tests :) 
```suggestion
    pub fn new(flow_alias: T, connector_alias: impl Into<String>) -> Self
    where
      T: Into<FlowAlias> {
        Self {
            flow_alias: flow_alias.into(),
            connector_alias: connector_alias.into(),
        }
    }
```

Reply:
done that with an `impl Into<FlowAlias>`

Reply:
Sorry, this was about something different, not that `From<>` etc exists  but that the `new` function tales a `Into<Flow Alias>` as as first argument.

It's also worth noting that there is a difference between a the generic version `T: Iinto...` and `impl Into<>` as one is compile time resolved the other at runtime, but I'm not sure it's really worth stressing out about perf in this instance OTOH it might be good not getting into the habit of `impl ...` if we don't have to

Reply:
as far as i know the `impl` in argument position is just syntactic sugar for a generic type param.
The reference says it:

> That is, impl Trait in argument position is syntactic sugar for a generic type parameter like <T: Trait>, except that the type is anonymous and doesn't appear in the [GenericParams](https://doc.rust-lang.org/reference/items/generics.html) list.

So I think it is not really handled at runtime, but the specialization during compile time is happening the same, isn't it?

Reply:
Also i put an `impl Into<FlowAlias>` into the first argument.